src/backend/routers/conversation.py

```diff
@@ -324,6 +329,89 @@ async def upload_file(
     return upload_file
 
 
+@router.post("/batch_upload_file", response_model=list[UploadFile])
+async def batch_upload_file(
+    session: DBSessionDep,
+    request: Request,
+    conversation_id: str = Form(None),
+    files: list[FastAPIUploadFile] = RequestFile(...),
+) -> UploadFile:
+    """
+    Uploads and creates a batch of File object.
+    If no conversation_id is provided, a new Conversation is created as well.
+
+    Args:
+        session (DBSessionDep): Database session.
+        file (list[FastAPIUploadFile]): List of files to be uploaded.
+        conversation_id (Optional[str]): Conversation ID passed from request query parameter.
+
+    Returns:
+        list[UploadFile]: List of uploaded files.
+
+    Raises:
+        HTTPException: If the conversation with the given ID is not found. Status code 404.
+        HTTPException: If the file wasn't uploaded correctly. Status code 500.
+    """
+
+    user_id = get_header_user_id(request)
+
+    validate_batch_file_size(session, user_id, files)
+
+    # Create new conversation
+    if not conversation_id:
+        conversation = conversation_crud.create_conversation(
+            session,
+            ConversationModel(user_id=user_id),
+        )
+    # Check for existing conversation
+    else:
+        conversation = conversation_crud.get_conversation(
+            session, conversation_id, user_id
+        )
+
+        # Fail if user_id is not provided when conversation DNE
+        if not conversation:
+            if not user_id:
+                raise HTTPException(
+                    status_code=400,
+                    detail=f"user_id is required if no valid conversation is provided.",
+                )
+
+            # Create new conversation
+            conversation = conversation_crud.create_conversation(
+                session,
+                ConversationModel(user_id=user_id),
+            )
+
+    # TODO: check if file already exists in DB once we have files per agents
+
+    # Handle uploading File
+    files_to_upload = []
+    for file in files:
```

Comment:
Any chance we can make this a streaming endpoint so we can return the status per file?